src/firefly/java/edu/caltech/ipac/firefly/util/ImageSetConverter.java

```diff
@@ -0,0 +1,209 @@
+package edu.caltech.ipac.firefly.util;
+
+import edu.caltech.ipac.firefly.server.util.DsvToDataGroup;
+import edu.caltech.ipac.firefly.server.util.Logger;
+import edu.caltech.ipac.util.DataGroup;
+import edu.caltech.ipac.util.DataObject;
+import edu.caltech.ipac.util.DataType;
+import edu.caltech.ipac.util.download.FailedRequestException;
+import org.apache.commons.csv.CSVFormat;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.List;
+
+
+/**
+ * Created by zhang on 9/27/17.
+ * This class will read in the ImageSet information table (maybe from an API) and
+ * the convert to a JSONObject.
+ */
+public class ImageSetConverter {
+    private  final String masterTablePath = "/edu/caltech/ipac/firefly/resources/";
+    private  final String irsaMasterTableName = "irsa-master-table.csv";
+    private  final String lsstMasterTableName = "lsst-master-table.csv";
+    private static final Logger.LoggerImpl LOG = Logger.getLogger();
+
+    private JSONArray imageSetToJSONArray =new JSONArray();
+    private static HashMap<String, String> paramMaps = new HashMap<>();
+
+    public ImageSetConverter(String whichProject) throws IOException, FailedRequestException {
+
+        //define the key mappings between the master table to the json file
+        paramMaps.put("apiType", "Service");
+        paramMaps.put("dataProductId", "surveyKey");
+        paramMaps.put("waveBandId", "surveyKeyBand");
+        paramMaps.put("waveBandDesc", "title");
+        paramMaps.put("filter", "filter");
+
+
+
+        String masterTable=null;
+        switch (whichProject.toLowerCase()){
+            case "irsa":
+                masterTable= new String(masterTablePath+irsaMasterTableName);
+                break;
+            case "lsst":
+                masterTable= new String(masterTablePath+lsstMasterTableName);
+                break;
+        }
+        imageSetToJSONArray = createImageJSONArray(masterTable);
+        //test the result
+     /*   try {
+            writeJSONArrayToFile(imageSetToJSONArray,  "/Users/zhang/IRSA_Dev/testingData/irsa-master-table.json");
```

Comment:
this can be removed before merging.

Reply:
Yes,  good idea.  I will move it to a test case.